src/Rings/MPolyQuo.jl

```diff
@@ -89,6 +89,10 @@ function Base.deepcopy_internal(a::MPolyQuoRingElem, dict::IdDict)
   return MPolyQuoRingElem(Base.deepcopy_internal(a.f, dict), a.P, a.simplified)
 end
 
+function is_zero(a::MPolyQuoRingElem)
+  return a.f in modulus(parent(a))
+end
```

Comment:
I seem to recall a recent pull request on this by @thofma ?

Reply:
Ah but this concerned rings which have a singular side?

Reply:
> `MPolyQuoRing` assumes that there is some kind of unique representative that we can compute (a la `simplify`).

I do not think, this is a reasonable assumption.

Reply:
> Then maybe `MPolyQuoRing` is not the right thing to use. Without `simplify`, it does not satisfy the ring interface, since arithmetic is not working with the current implementation.

I'm afraid, I'm missing some discussions that you have had with others to reach these conclusions. So I don't really understand what the problem is. 

What it looks like from my perspective: Most probably, `simplify` (or `simplify!`) is supposed to pick a distinguished representative for every element. This is usually achieved by choice of a monomial ordering when working over a reasonable field of coefficients. 

The arithmetic in quotient rings requires an equality check for elements (`==`). If a normal form is provided via `simplify`, one can use that to decide equality. But mathematically, the equality check is independent of that and can also be carried out by testing ideal membership (as done in the generic code I propose here). For the coefficient rings I am interested in there probably is no reasonable meaning of `simplify` and even if there was, we would certainly lack the singular backend to implement it. 

So far, we wanted to model the mathematical objects in Oscar with no additional "non-mathematical" information and requirements; the prominent example being that polynomial rings do not have an ordering. In this sense it seems to me that requiring `simplify` to be implemented for quotient rings in general is breaking with this convention. Or am I understanding something wrong?

Reply:
> I don't like that we remove the `simplify` from some functions to make them unnecessarily slower (which there were complaints about), but keep them for others. It is just an inconsistent mess.

I see the point about the inconsistency. But I am keeping the simplify functionality for all the relevant cases where it actually has a computational backend. And there your recent changes are an improvement! But why not make a different dispatch for the exotic remaining ones? That method could also make use of the `a.simplified` field for the `iszero` check but using ideal membership instead. But it would have to be well documented how the value of this field must be interpreted in this general case. 

Be aware that my last commits were taking care of the duplicate method. This looks different now.

Reply:
I just pushed an example of what I had in mind with my last message.

Reply:
Every model assumes something. So far the `MPolyQuoRing` was assuming, that one has a way of obtaining a unique representative (as is evident by the usage of `simplify`). This makes them performant in many situations. Is this the only way to model quotient rings? Surely no, there is also a model for quotients of Euclidean domains, which is even faster, but of course assumes more. I don't see why this is breaking with the convention of modelling mathematics. We are not claiming to model everything with the same underlying representation. For a specific ring you might get a other implementation. But a quotient ring constructor will yield a quotient ring.

Please correct me if I am wrong, but it seems that you want yet another model for a quotient, which is the most generic one and at the same time less performant one. One models `R/I` by working in `R` with the additional `==(x, y) = (x - y) in I` method. It does not matter whether it is a quotient of a multivariate polynomial ring.

What I am not comfortable with is changing `MPolyQuoRing` to be less and less useful for the rings it was intended to by making it more and more generic. Thanks for the latest commit. I am happy with using dispatch for now. It will be a bit awkward if one wants to change all the functions that currently use `simplify`, but this should not matter for now.

Reply:
I see your point about making some basic assumptions when designing a new type. Hence, I also understand objections to my overriding the generic methods for `iszero` and `simplify`, especially right after you had polished them. Still, it would be nice if I could make use of the existing types and accompanying infrastructure for my purposes. Maybe, it is then better that I do special method dispatch for my particular cases and leave the generic code as it was. I'll try to be concise and explain what I do in comments.

Reply:
I think your changes are fine. I am happy if we add it in this form.